asf_heat_pump_suitability/analysis/exploratory/create_full_dataset/create_full_dataset_plymouth.py

```diff
@@ -0,0 +1,1182 @@
+        ).alias("prop_disabled_residents")
+    )
+    .select(["oa_code", "prop_disabled_residents"])
+)
+
+# Process economic status data
+economic_status_df = (
+    economic_status_df.filter(
+        pl.col("2021 output area").is_not_null(), pl.col("2021 output area") != ""
+    )
+    .rename({"2021 output area": "oa_code"})
+    .with_columns(
+        (
+            pl.col("Economically inactive")
+            / pl.col("Total: All usual residents aged 16 years and over")
+        ).alias("prop_economically_inactive_adults")
+    )
+    .select(["oa_code", "prop_economically_inactive_adults"])
+)
+
+# Process hours worked data
+hours_worked_df = (
+    hours_worked_df.filter(
+        pl.col("2021 output area").is_not_null(), pl.col("2021 output area") != ""
+    )
+    .rename({"2021 output area": "oa_code"})
+    .with_columns(
+        (
+            pl.col("Full-time")
+            / pl.col(
+                "Total: All usual residents aged 16 years and over in employment the week before the census"
+            )
+        ).alias("prop_employed_adults_working_full_time")
+    )
+    .select(["oa_code", "prop_employed_adults_working_full_time"])
+)
+
+# Process student binary
+student_binary_df = (
+    student_binary_df.filter(
+        pl.col("2021 output area").is_not_null(), pl.col("2021 output area") != ""
+    )
+    .rename({"2021 output area": "oa_code"})
+    .with_columns((pl.col("Student") / pl.col("Total")).alias("prop_students"))
+    .select(["oa_code", "prop_students"])
+)
+
+# Process tenure
+oa_tenure_df = (
+    oa_tenure_df.filter(
+        pl.col("2021 output area").is_not_null(), pl.col("2021 output area") != ""
+    )
+    .rename({"2021 output area": "oa_code"})
+    .select(
+        [
+            "oa_code",
+            "Total: All households",
+            "Owned",
+            "Shared ownership",
+            "Social rented",
+            "Private rented",
+            "Lives rent free",
+        ]
+    )
+    .with_columns(
+        (
+            pl.sum_horizontal("Owned", "Shared ownership")
+            / pl.col("Total: All households")
+        ).alias("prop_owner_occupied"),
+        (pl.col("Social rented") / pl.col("Total: All households")).alias(
+            "prop_social_rented"
+        ),
+    )
+    .select(["oa_code", "prop_owner_occupied", "prop_social_rented"])
+)
+
+# Join census datasets together
+census_df = oa_tenure_df
+for df in [
+    age_bands_df,
+    disability_df,
+    economic_status_df,
+    hours_worked_df,
+    student_binary_df,
+]:
+    census_df = census_df.join(df, how="left", on="oa_code")
+
+# %%
+# CREATE DATAFRAME WITH FEATURES TO TRAIN AND TEST MODEL
+model_df = (
+    epc_df.select(
+        [
+            "UPRN",
+            "property_type",
+            "TENURE",
+            "oa21",
+        ]
+    )
+    .join(census_df, how="left", left_on="oa21", right_on="oa_code")
+    .join(
+        features_df.select(["UPRN", "in_cons_area", "in_listed_building"]),
```

Comment:
```suggestion
        features_df.select(["UPRN", "in_conservation_area", "in_listed_building"]),
```
if you decide to change above